server/embed/etcd.go

```diff
@@ -774,3 +797,29 @@ func parseCompactionRetention(mode, retention string) (ret time.Duration, err er
 	}
 	return ret, nil
 }
+
+func (e *Etcd) setupTracing(ctx context.Context) (tracesdk.SpanExporter, error) {
+	exporter, err := otlp.NewExporter(ctx,
+		otlpgrpc.NewDriver(
+			otlpgrpc.WithEndpoint(e.cfg.ExperimentalDistributedTracingAddress),
+			otlpgrpc.WithInsecure(),
+		))
+	if err != nil {
+		return nil, err
+	}
+	res := resource.NewWithAttributes(
+		semconv.ServiceNameKey.String(e.cfg.ExperimentalDistributedTracingServiceName),
+	)
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithSyncer(exporter),
```

Comment:
WithSyncer is synchronous (blocks until the trace has been sent).  See https://github.com/open-telemetry/opentelemetry-go/blob/24a91628f112f234af024afb42be57247752bbd0/sdk/trace/provider.go#L240.  You should use WithBatcher instead.